src/main/java/org/candlepin/subscriptions/files/ProductProfile.java

```diff
@@ -140,6 +171,17 @@ public boolean supportsGranularity(Granularity granularity) {
         return granularity.compareTo(finestGranularity) < 1;
     }
 
+    public Map<String, Set<String>> mapSwatchProductsByRole() {
+        return Optional.ofNullable(this.getSyspurposeRoles()).orElse(Collections.emptySet()).stream()
+            .collect(Collectors.toMap(SyspurposeRole::getName, SyspurposeRole::getSwatchProductIds));
+    }
+
+    public Map<String, Set<String>> mapSwatchProductsByEngProducts() {
+        return Optional.ofNullable(this.getProducts()).orElse(Collections.emptySet()).stream()
+            .collect(Collectors.toMap(SubscriptionWatchProduct::getEngProductId,
+                SubscriptionWatchProduct::getSwatchProductIds));
+    }
+
```

Comment:
Rather than compute these maps every time, is there a reason not to just go ahead and store the maps as a field in this class (maybe as a side-effect of `setProducts` and `setSyspurposeRoles()`, ensuring we use those in the constructor?)

Reply:
:+1:  Good idea.

Reply:
DONE